dds_web/api/user.py

```diff
@@ -45,33 +45,42 @@ class AddUser(flask_restful.Resource):
     @auth.login_required(role=["Super Admin", "Unit Admin", "Unit Personnel", "Project Owner"])
     @logging_bind_request
     def post(self):
-        """Create an invite and send email."""
+        """Associate existing users or unanswered invites with projects or create invites"""
 
+        # A project may or may not be specified
         project = flask.request.args.get("project", None)
+        if project:
+            project = project_schemas.ProjectRequiredSchema().load({"project": project})
+            role = flask.request.json.get("role", None)
+
         args = flask.request.json
         # Check if email is registered to a user
         existing_user = user_schemas.UserSchema().load(args)
+        unanswered_invite = user_schemas.UnansweredInvite().load(args)
 
-        if existing_user and not project:
-            raise ddserr.DDSArgumentError(
-                message="User exists! Specify a project if you want to add this user to a project."
-            )
-
-        if not existing_user:
-            # Send invite if the user doesn't exist
-            invite_user_result = self.invite_user(args)
-            return invite_user_result, invite_user_result["status"]
+        if existing_user or unanswered_invite:
```

Comment:
Should we have a check to make sure we don't have both `existing_user` and `unanswered_invite`. That shouldn't be allowed to happen, but doesn't hurt to check for it I guess?

Reply:
Added suggestion above.

Reply:
Thank you for the suggestion. In the Unanswered Invite Schema, I have now made sure that the Invite Object is only returned if no user with that email exists:
```
        # double check if there is no existing user with this email
        userexists = utils.email_in_db(email=data.get("email"))

        if userexists and invite:
            raise ddserr.DatabaseError(message="Email exists for user and invite at the same time")

        # if the user exists already, the invite object must not be returned to prevent sign-up
        return invite if not userexists else None
```

I feel this is the better place for it, even though we indeed don't use that schema elsewhere, so we could have it also in the API endpoint code directly.

Reply:
Yes, that's probably a better place for it! 👍